cns/restserver/metrics.go

```diff
@@ -219,7 +219,7 @@ func (a *asyncMetricsRecorder) record() {
 func (service *HTTPRestService) publishIPStateMetrics() {
 	recorder.once.Do(func() {
 		recorder.podIPConfigSrc = service.PodIPConfigStates
-		recorder.sig = make(chan struct{})
+		recorder.sig = make(chan struct{}, 1)
```

Comment:
disagree with this change, unless this can't be called more than twice concurrently (and if that was the case, why twice instead of once and leave this unbuffered?). this smells and indicates some other structural problem.
all it does is _delay_ when you will get a blocking send by one more event. maybe sig should only have non-blocking sends?